share/tunnel/tunnel_in_proxy.go

```diff
@@ -52,6 +58,21 @@ func (p *Proxy) listen() error {
 		if err != nil {
 			return p.Errorf("tcp: %s", err)
 		}
+		if p.remote.Socks {
+			udpAddr, err := net.ResolveUDPAddr("udp", p.remote.LocalHost+":0")
+			if err != nil {
+				_ = l.Close()
+				return p.Errorf("resolve: %s", err)
+			}
+
+			sl := log.New(ioutil.Discard, "", 0)
+			if p.Logger.Debug {
+				sl = log.New(os.Stdout, "[socks]", log.Ldate|log.Ltime)
+			}
+			p.socksServer, _ = socks5.New(&socks5.Config{
+				Handler: makeChiselSocksHandler(p, udpAddr, sl),
```

Comment:
If this code does belong in Proxy, then Instead of using `sl` we should use `p.Logger.Fork("socks")`